### PR TITLE
Add basic maximum payment value protections in tooling and enclave

### PR DESCRIPTION
**Pull request by Sneagan**
### Type of Change

- [ ] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before Requesting Review

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security and/or privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: tools/payments/report.go [hunk 1]
```diff
@@ -235,6 +235,10 @@ func (r PreparedReport) Prepare(ctx context.Context, key ed25519.PrivateKey, cli
 
 	reqs := make([]payments.PrepareRequest, len(r))
 	for i, paymentDetails := range r {
+		// Implement basic maximum payout amount protection
+		if paymentDetails.Amount.GreaterThan(decimal.NewFromInt32(50)) {
```
Comment (pavelbrm) [inline tools/payments/report.go:239]: Nit: the limit could be moved up outside of the loop.

**Review by pavelbrm — APPROVED** [review]

File: tools/payments/report.go (hunk 1 above)
Comment (Sneagan) [reply tools/payments/report.go:239]: Realized we should reuse the value defined in `clients` until we make it configurable.

File: libs/clients/client.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
 )
 
 // TransferLimit is the limit of BAT any client can transfer
-var TransferLimit = decimal.NewFromInt(300)
```
Comment (evq) [inline libs/clients/client.go:27]: :sob: why is this here of all places
